roles/confluent.kafka_connect/tasks/main.yml

```diff
@@ -103,3 +103,10 @@
     name: "{{kafka_connect.service_name}}"
     enabled: "{{kafka_connect.systemd.enabled}}"
     state: "{{kafka_connect.systemd.state}}"
+
+- name: Register connector configs and remove deleted connectors
+  confluent_connectors:
+    connect_url: "{{kafka_connect_http_protocol}}://{{groups['kafka_connect'][0]}}:{{kafka_connect_rest_port}}/connectors"
+    active_connectors: "{{ kafka_connect_connectors }}"
+  when: kafka_connect_connectors is defined and groups['kafka_connect']|count > 0
```

Comment:
remove the `and groups['kafka_connect']|count > 0` if there were no hosts under kafka_connect then this role wouldnt get called